domain/port/service/watcher.go

```diff
@@ -59,20 +61,16 @@ type WatcherState interface {
 	// event for the WatchMachineOpenedPorts watcher
 	InitialWatchMachineOpenedPortsStatement() string
 
-	// InitialWatchApplicationOpenedPortsStatement returns the query to load the
-	// initial event for the WatchApplicationOpenedPorts watcher
-	InitialWatchApplicationOpenedPortsStatement() string
-
 	// GetMachineNamesForUnitEndpoints returns map from endpoint uuids to the uuids of
 	// the machines which host that endpoint for each provided endpoint uuid.
 	GetMachineNamesForUnitEndpoints(ctx context.Context, endpointUUIDs []string) ([]coremachine.Name, error)
 
-	// GetApplicationNamesForUnitEndpoints returns a slice of application names that
-	// host the provided endpoints.
-	GetApplicationNamesForUnitEndpoints(ctx context.Context, eps []string) ([]string, error)
+	// FilterEndpointsForApplication returns the subset of provided endpoint uuids
+	// that are associated with the provided application.
+	FilterEndpointsForApplication(ctx context.Context, app coreapplication.ID, eps []string) (set.Strings, error)
```

Comment:
**nitpick:** `FilterXForY(ctx, x, y)` would be more logical (here we have `FilterXForY(ctx, y, x)`)

**suggestion:** `FilterEndpointsForApplication(ctx context.Context, eps []string, app coreapplication.ID) (set.Strings, error)`